### PR TITLE
gh-121905: Consistently use "floating-point" instead of "floating point"

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-121905

----
📚 Documentation preview 📚: https://cpython-previews--121907.org.readthedocs.build/

### CODE REVIEW

**Comment by serhiy-storchaka** [comment]
I tried to only use "floating-point" for adjectives, like "floating-point number" and "floating-point arithmetic", not for nouns, like "binary floating point" and "decimal floating point", and not change book names, but it needs third party verification. Maybe there are other cases in which the hyphen is not desirable.

File: Doc/c-api/float.rst [hunk 1]
```diff
@@ -2,20 +2,20 @@
 
 .. _floatobjects:
 
-Floating Point Objects
+Floating-Point Objects
 ======================
 
-.. index:: pair: object; floating point
+.. index:: pair: object; floating-point
```
Comment (AA-Turner) [inline Doc/c-api/float.rst:8]: The [documentation index](https://cpython-previews--121907.org.readthedocs.build/en/121907/genindex-F.html) now has entries for *floating point* and *floating-point*, which seems unhelpful -- the remaining non-hyphenated index entries could also be changed?

**Review by AlexWaygood — APPROVED** [review]

**Comment by AlexWaygood** [comment]
> I tried to only use "floating-point" for adjectives, like "floating-point number" and "floating-point arithmetic", not for nouns, like "binary floating point" and "decimal floating point", and not change book names, but it needs third party verification. Maybe there are other cases in which the hyphen is not desirable.

This is the correct distinction; your choices all look correct to me

File: Doc/faq/programming.rst [hunk 2]
```diff
@@ -869,7 +869,7 @@ How do I convert a string to a number?
 --------------------------------------
 
 For integers, use the built-in :func:`int` type constructor, e.g. ``int('144')
-== 144``.  Similarly, :func:`float` converts to floating-point,
+== 144``.  Similarly, :func:`float` converts to floating point,
 e.g. ``float('144') == 144.0``.
```
Comment (AA-Turner) [inline Doc/faq/programming.rst]: Perhaps? (decimal/number/value/etc)
```suggestion
== 144``.  Similarly, :func:`float` converts to a floating-point decimal,
e.g. ``float('144') == 144.0``.
```

File: Doc/reference/datamodel.rst [hunk 3]
```diff
@@ -263,17 +263,19 @@ Booleans (:class:`bool`)
 
 .. index::
    pair: object; floating point
+   pair: object; floating-point
    pair: floating point; number
+   pair: floating-point; number
```
Comment (AA-Turner) [inline Doc/reference/datamodel.rst]: I would suggest removing the unhyphenated per the above:
```suggestion
   pair: object; floating-point
   pair: floating-point; number
```

File: Doc/tutorial/floatingpoint.rst [hunk 4]
```diff
@@ -6,7 +6,7 @@
 .. _tut-fp-issues:
 
 **************************************************
-Floating Point Arithmetic:  Issues and Limitations
+Floating-Point Arithmetic:  Issues and Limitations
 **************************************************
```
Comment (AA-Turner) [inline Doc/tutorial/floatingpoint.rst:10]: Capitalisation seems odd here with the hyphen, but a minor point.

**Review by AA-Turner — APPROVED** [review]
I've had a brief look through everything and can't spot any obvious problems. I agree re using the hyphenated form.

A

File: Doc/reference/datamodel.rst (hunk 3 above)
Comment (ezio-melotti) [reply Doc/reference/datamodel.rst]: If these affect searching, it seems more likely that people will search for the non-hyphenated version, so I would keep that instead.

File: Doc/tutorial/floatingpoint.rst (hunk 4 above)
Comment (ezio-melotti) [reply Doc/tutorial/floatingpoint.rst:10]: This should use [sentence case](https://devguide.python.org/documentation/style-guide/#capitalization) anyway, but the whole document should be updated to keep it consistent.
There's also a double space after the `:` that seems out of place.

File: Doc/faq/programming.rst (hunk 2 above)
Comment (serhiy-storchaka) [reply Doc/faq/programming.rst]: Why decimal? It is a binary floating point.

File: Doc/tutorial/floatingpoint.rst (hunk 4 above)
Comment (serhiy-storchaka) [reply Doc/tutorial/floatingpoint.rst:10]: I leave this to other issue then.

File: Doc/c-api/float.rst (hunk 1 above)
Comment (serhiy-storchaka) [reply Doc/c-api/float.rst:8]: i was not sure what should be in the index. Removing the non-hyphenated index entries then.

File: Doc/faq/programming.rst (hunk 2 above)
Comment (AA-Turner) [reply Doc/faq/programming.rst]: "value" or "number" would be equally fine, the suggestion was more adding some sort of noun -- you are right that "decimal" could be confusing here.

**Review by mdickinson — APPROVED** [review]
Haven't reviewed line-by-line, but the change LGTM in principle.

**Pull request merged**